### DISCUSSION TITLE
Slint UI is one of the Robust - what we require to become like flutter and better than flutter

### DISCUSSION THREAD

**Discussion by MohaBeacon** [slint-ui/slint#2593] [Ideas]
Slint UI - on microcontroller is something special. What I personally wish the future could be the library integration from Rust or other languages inside Slint UI. For example, Dashboard has different formats, camera have different format as output, from low end devices like Bluetooth integration in UI

**Comment by tronical** [comment]
Hi! I'm not sure I understand entirely. What do you mean with "library integration from Rust or other languages inside Slint UI" ? Do you mean embedding Rust code in .slint files?

**Reply by MohaBeacon** [reply]
i2c / spi component with data format - just like rectange - horizonal/vertical - which will process and render out data for the display inside slint. Bcoz the UI which supports data format will be better to handle will become trust worthy

**Reply by tronical** [reply]
Thank you for clarifying - now I see what you mean. At this point we don't have any plans to expand the scope of the language beyond user interfaces.
